pkg/entrypoint/entrypointer.go

```diff
@@ -143,21 +163,25 @@ func (e Entrypointer) Go() error {
 		ResultType: result.InternalTektonResultType,
 	})
 
-	ctx := context.Background()
 	var err error
-
 	if e.Timeout != nil && *e.Timeout < time.Duration(0) {
 		err = fmt.Errorf("negative timeout specified")
 	}
-
+	ctx := context.Background()
+	var cancel context.CancelFunc
 	if err == nil {
-		var cancel context.CancelFunc
-		if e.Timeout != nil && *e.Timeout != time.Duration(0) {
+		ctx, cancel = context.WithCancel(ctx)
+		if e.Timeout != nil && *e.Timeout > time.Duration(0) {
 			ctx, cancel = context.WithTimeout(ctx, *e.Timeout)
-			defer cancel()
 		}
+		// start a goroutine to listen for cancellation file
+		go func() {
```

Comment:
Is it possible that this goroutine would leak if the taskrun is never canceled? i.e. the goroutine is waiting for a downwardapi file to exist; if it never does, I'm not sure whether the goroutine will terminate.

Reply:
Thank you for your reminder, I added defer cancel(), which will not affect the cancellation function, but also allow the goroutine to be released